src/main/scala-sbt-1.0/sbtprotoc/Compat.scala

```diff
@@ -14,9 +12,7 @@ private[sbtprotoc] trait Compat extends CacheImplicits { self: ProtocPlugin.type
   }
 
   protected object CacheArguments {
-    implicit val instance: JsonFormat[Arguments] = project(
-      (a: Arguments) => (a.includePaths, a.protocOptions, a.pythonExe, a.deleteTargetDirectory, a.targets),
-      (in: (Seq[File], Seq[String], String, Boolean, Seq[(File, Seq[String])])) => Arguments(in._1, in._2, in._3, in._4, in._5)
-    )
+    implicit val instance: JsonFormat[Arguments] =
+      caseClassArray(Arguments.apply _, Arguments.unapply _)
```

Comment:
One of the reasons it was using tuples is because there's a bug with HList serialization/deserialization + the use of `previous`. This may be the root of your problem @dwijnand.